jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsResourceCache.kt

```diff
@@ -291,12 +294,25 @@ class DefaultAwsResourceCache(
 
     private fun <T> getCachedResource(context: Context<T>) {
         ApplicationManager.getApplication().executeOnPooledThread {
+            var currentValue: Entry<T>? = null
             try {
                 @Suppress("UNCHECKED_CAST")
                 val result = cache.compute(context.cacheKey) { _, value ->
-                    fetchIfNeeded(context, value as Entry<T>?)
+                    currentValue = value as Entry<T>?
+                    fetchIfNeeded(context, currentValue)
                 } as Entry<T>
-                context.future.complete(result.value)
+                runBlocking {
```

Comment:
Why does this need to be runBlocking?

Reply:
it doesn't; there were some issues with bubbling exceptions since it needs to be catch `Throwable` instead of `Exception`